Allow multiple package specifiers at the same time

ahal opened the pull request:
This allows e.g:
    $ hashin packageA==1.2.3 packageB==2.3.4 packageC

This change means that 'requirements_file' and 'algorithm' can
no longer be positional arguments, so -r/--requirements-file and
-a/--algorithm were added for them respectively. Obviously this
breaks backwards compat.

This is also the simplest way to implement it, but not necessarily
the most efficient. For example, this will open/close the requirements
file each time, rather than leaving it open for the duration.

peterbe commented:
Yay! I feel so bad for making the requirements.txt and algorithm positional arguments. 
When time allows I'll give it a proper review. I'm more than happy to up the major number after this is landed.

peterbe commented:
I pushed [0.6.1 to master moments ago](https://github.com/peterbe/hashin/commit/2df67ca0c9fda3bb0222503a2f04f38be2ee743b). Would you mind rebasing this branch and add a blob to the README about what a 1.0.0 version would be. I.e. the discussion above about multiple packages and needing to keyword argument the requirements file and algorithm.

peterbe commented:
By the way, don't worry about closing and re-opening the requirements file for each package. It takes nanoseconds for python to do that. 

The only thing to consider is atomicity. I.e. if I do `hashin requests junkcrap` should it really have added `requests` before it gets an error on `junkcrap`? After all, if you do `pip install requests junkcrap` it will, I believe, install `requests` first and then try (and fail) to install `junkcrap`. 

Perhaps best to not worry about it for now. It's cheap if hashin has to add something that it already has.

ahal commented:
I rebased and added a blurb to the README. I agree that it would be nice if it detected dependencies and didn't update the requirements file more than one. But that's a bit more work than I have time for :)

peterbe commented on README.rst:
```diff
@@ -77,13 +77,22 @@ Or you can be specific about exactly which version you want::
-directory you can do this::
+directory you can specify ``--requirements-file``::
+
+    hashin futures --requirements-file stuff/requirements/prod.txt
```
Perhaps I'm old-school but I think it always looks better to use a `=` with keyword arguments. Except if it's a single key one. E.g.
```
hashin futures --requirements-file=stuff/requirements/prod.txt
hashin futures --algorithm=sha512
```
I'm pretty sure argparse can cope with both but this style without the `=` looks so odd to me :)

peterbe commented on hashin.py:
```diff
@@ -284,41 +284,45 @@ def main():
+
+    ret = 0
+    for package in args.packages:
+        ret = ret | run(
```
Too much logic here in the `if __name__==__main__` block thing. Can you change `run()` to take a list instead. 
And it, should return 0 if ALL worked. But if there's any exceptions (e.g. HTTP 404 on an unknown package the exception itself will take care of making sure it doesn't exit with a 0.

peterbe commented:
See https://github.com/peterbe/hashin/issues/26
I'm very eager to fix that but I'd like to merge yours first. Perhaps it's not related but it just smells like it might clash.

peterbe commented:
Wonderful! I'm going to merge it and push out a new release and make a big stink about it on twitter and blogpost etc.

peterbe commented:
https://www.peterbe.com/plog/hashin-0.7.0
